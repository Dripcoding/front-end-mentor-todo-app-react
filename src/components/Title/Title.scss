@import "../../mixins";

.title__container {
  @include flex(row, space-between, flex-start);
  width: 540px;
  height: 64px;

  h1 {
    @include text(white, 40px, normal, 700, 15px, 40px, left, capitalize);
    width: 167px;
    height: 40px;
  }

  svg {
    align-self: center;
    margin-top: 20px;
  }
}