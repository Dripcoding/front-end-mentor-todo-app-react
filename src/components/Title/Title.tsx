import React from "react";

import { ReactComponent as SunToggleImg } from "../../assets/images/icon-sun.svg";
import "./Title.scss";

const STYLE_BASE: string = "title__";

const Title = (): JSX.Element => {
  return (
    <div className={`${STYLE_BASE}titleContainer`}>
      <h1>Todo</h1>
      <SunToggleImg data-testid="SUN-TOGGLE-ICON" />
    </div>
  );
};

export default Title;