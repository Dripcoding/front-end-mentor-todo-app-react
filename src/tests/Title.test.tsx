import { render } from "@testing-library/react";
import Title from "../components/Title/index";


describe("Title", () => {
  it("matches snapshot", () => {
    const { container } = render(<Title />);
    expect(container).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { getByRole, getByTestId } = render(<Title />);

    const todoHeader = getByRole("heading");
    expect(todoHeader).toBeInTheDocument();
    expect(todoHeader).toBeVisible();

    const sunIcon = getByTestId("SUN-TOGGLE-ICON");
    expect(sunIcon).toBeInTheDocument();
    expect(sunIcon).toBeVisible();
  });
});