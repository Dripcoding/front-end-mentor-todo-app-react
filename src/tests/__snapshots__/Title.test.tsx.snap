// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title matches snapshot 1`] = `
<div>
  <div
    class="title__container"
  >
    <h1>
      Todo
    </h1>
    <svg
      data-testid="SUN-TOGGLE-ICON"
    >
      icon-sun.svg
    </svg>
  </div>
</div>
`;
